librecad/src/actions/rs_actionpolylinesegment.cpp

```diff
@@ -54,9 +54,9 @@ void RS_ActionPolylineSegment::init(int status) {
 	if (container->countSelected(true, entityType)) {
 		//find a selected entity
 		//TODO, find a better starting point
-		for (RS_Entity* e = container->firstEntity(RS2::ResolveAllButInserts);
+		for (RS_Entity* e = container->firstEntity(RS2::ResolveNone);
```

Comment:
I think it can't be elliptical arcs because the arcs in polylines are represented with the single 'bulge' coordinate in the DXF format, not with real objects like in LibreCAD. I've ran into this problem before when trying to scale polyline with radiuses with different X and Y scale - the drawing fell apart, and the reason why was: it would turn the arcs into elliptical arcs, and that's not supported in polylines.

But, I think, this particular loop is over the objects in the whole drawing, not in a polyline. Previously, all polylines in the drawing were resolved, which caused unwanted effects: imagine the situation where the selected object is a polyline, but while iterating it's resolved into its components. So, the final list of segments would contain both the polyline as a whole object (added outside of this loop) and the components (added by this loop). That caused the resulting polyline to contain the same segments twice, hence the bug report.

This change means that the polylines will be treated as "opaque" objects with only start and end. This is okay because in LibreCAD one can only select a polyline in its entirety, not the lines/arcs that it's made of. So, the usable selected objects for this operation will be either lines, arcs or polylines.

Maybe there's a version of polyline object in DXF with separate x/y radius? I'm not very familiar with the format.

Hope the explanation makes sense - I'm not good at putting this in words.